### PR TITLE
add dim/BASIC

### PR DESCRIPTION
**Pull request by jeroenvollenbrock**
added the BASIC command class for the dim capability, as BASIC can be
used for dimming purposes (0 - 99)

### CODE REVIEW

**Comment by jeroenvollenbrock** [comment]
extracted from #44

File: lib/zwave/system/capabilities/dim/BASIC.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	get: 'BASIC_GET',
+	set: 'BASIC_SET',
+	setParser(value) {
+		if (this.hasCapability('onoff')) this.setCapabilityValue('onoff', value > 0);
+		return {
+			Value: Math.round(value * 99),
+		};
+	}),
+	report: 'BASIC_REPORT',
+	reportParser: report => {
```
Comment (RobinBol) [inline lib/zwave/system/capabilities/dim/BASIC.js]: I don't think this works, did you test it? The scope in the `reportParser` is not correct, `this` is not the device instance. You need to format your function similar to the `setParser`.

File: lib/zwave/system/capabilities/dim/BASIC.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		if (report && report.hasOwnProperty('Value')) {
+			if (this.hasCapability('onoff')) this.setCapabilityValue('onoff', value > 0);
```
Comment (RobinBol) [inline lib/zwave/system/capabilities/dim/BASIC.js]: `value` is undefined in this case?

File: lib/zwave/system/capabilities/dim/BASIC.js (hunk 1 above)
Comment (caseda) [reply lib/zwave/system/capabilities/dim/BASIC.js]: mmm i did not test when added in the system capabilities.
does work in my testing code for one of the devices i've worked on a little though still without setting onoff:
https://github.com/caseda/com.hank-tech/blob/temp_color_switch/drivers/HKZW-RGB01/device.js#L31
so you might be right that `this` is not going to be in the scope now

**Comment by RobinBol** [comment]
Merged the other one!

**Pull request closed without merging**